draft-barnes-acme.md

```diff
@@ -198,324 +212,362 @@ To revoke a certificate, the client simply sends a revocation request, signed wi
 
-jwk (required, object):
-: A JSON Web Key object describing the key used to verify the signature {{I-D.ietf-jose-json-web-key}}.
 
-Each usage of a signature object must specify the content being signed.  To avoid replay risk, the input to the signature algorithm is the concatenation of the nonce with the content to be signed.
+The only type of identifier defined by this specification is a fully-qualified domain name (type: "dns").  The value of the identifier MUST be the ASCII representation of the domain name.
 
 ~~~~~~~~~~
 
-      signature-input = nonce || content
+{
+  "status": "valid",
+  "expires": "2015-03-01",
 
-~~~~~~~~~~
+  "identifier": {
+    "type": "domain",
+    "value": "example.org"
+  },
 
-A verifier computes the same input before verifying the signature.  Note that while an signature object contains all of the information required to verify the signature, the verifier must also check that the public key encoded in the JWK object is the correct key for a given context.
+  "key": { /* JWK */ },
 
-## Key Authorization
+  "challenges": [
+    "simpleHttps": {
+      "status": "valid",
+      "validated": "2014-12-01T12:05Z",
+      "token": "IlirfxKKXAsHtmzK29Pj8A"
+      "path": "Hf5GrX4Q7EBax9hc2jJnfw"
+    },
+    "recoveryToken": {
+      "status": "valid",
+      "validated": "2014-12-01T12:07Z",
+      "token": "23029d88d9e123e"
+    }
+  ],
+}
 
-The key authorization process establishes a key pair as an authorized key pair for a given identifier.  This process must assure the server of two things: First, that the client controls the private key of the key pair, and second, that the client holds the identifier in question.  This process may be repeated to associate multiple identifiers to a key pair (e.g., to request certificates with multiple identifiers), or to associate multiple key pairs with an identifier (e.g., for load balancing).
+~~~~~~~~~~
 
-As illustrated by the figure in the overview section above, the authorization process proceeds in two transactions.  The client first requests a list of challenges from the server, and then requests authorization based on its answers to those challenges.
 
-The first request in the key authorization process is a "challengeRequest" message, specifying the identifier for which the client will be requesting authorization.  The fields in a "challengeRequest" message are as follows:
+## Key Authorization
 
-type (required, string):
-: "challengeRequest"
+The key authorization process establishes the authorization of an account key pair to manage certificates for a given identifier.  This process must assure the server of two things: First, that the client controls the private key of the key pair, and second, that the client holds the identifier in question.  This process may be repeated to associate multiple identifiers to a key pair (e.g., to request certificates with multiple identifiers), or to associate multiple key pairs with an identifier (e.g., for load balancing).
```

Comment:
I think the "for load balancing" example doesn't make sense, since the account key pair should not be the same as the certificate key pair. https://github.com/letsencrypt/acme-spec/issues/12